pkg/loop/internal/pb/chain_reader.proto

```diff
@@ -34,3 +62,133 @@ message BoundContract {
   string name = 2;
   bool pending = 3;
```

Comment:
We still need to update this to be finalized or confidence, right?

Reply:
Yes, this will be in a separate PR to make this easier to review

Reply:
I started throwing everything into this PR and it became a mess